decidim-assemblies/app/permissions/decidim/assemblies/permissions.rb

```diff
@@ -172,6 +173,20 @@ def user_can_read_assembly_list?
         toggle_allow(user.admin? || has_manageable_assemblies?)
       end
 
+      # The list of assemblies the user is able to list
+      # In case of user being admin of child assembly even parent assembly
+      # should be listed to be able to navigate to child assembly
+      def user_can_list_assembly_list?
+        return unless permission_action.action == :list &&
+                      permission_action.subject == :assembly
+
+        assemblies = AssembliesWithUserRole.for(user)
+        parent_assemblies = Decidim::Assembly.where(id: assemblies.pluck(:parent_id))
```

Comment:
This would only work for parent-child, but there's no limit in depth for assemblies, so you could have this scenario:

```
Assembly A
|- Assembly B
  |- Assembly C
```

The current code would not allow admins of Assembly C to list Assembly A.

I would change this to use the `ancestors` method defined in the `Assembly` model:

```ruby
assemblies_ids = AssembliesWithUserRole.for(user)
parent_assemblies = assemblies.flat_map{ |assembly| [assembly.id] + assembly.ancestors.pluck(:id) }

allowed_list_of_assemblies = assemblies + parent_assemblies
toggle_allow(allowed_list_of_assemblies.member?(assembly.id))
```

By using `pluck` we fetch only that column from the DB, so we avoid requiring extra data we won't use and we can compare IDs directly.

(You should double-check that code because I didn't check it worked properly!)